### DISCUSSION TITLE
Camera zoom issue

### DISCUSSION THREAD

**Discussion by mylaimani** [pedroSG94/RootEncoder#1579] [General]
My code is like this

        int iZoomLevel = 10;
        cameraZoomControls = findViewById(R.id.cameraZoomControls);
        cameraZoomControls.setActivated(true);
        cameraZoomControls.setIsZoomInEnabled(true);
        cameraZoomControls.setIsZoomOutEnabled(true);

        // Handle zoom-in click
        cameraZoomControls.setOnZoomInClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                iZoomLevel = iZoomLevel + 5;
                rtmpCamera1.setZoom(iZoomLevel);
                Toast.makeText(getApplicationContext(), "Zoom In Clicked!", Toast.LENGTH_SHORT).show();
            }
        });

       // Handle Zoom Out Click
        cameraZoomControls.setOnZoomOutClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                iZoomLevel = iZoomLevel - 5;
                rtmpCamera1.setZoom(iZoomLevel);
                Toast.makeText(getApplicationContext(), "Zoom Out Clicked!", Toast.LENGTH_SHORT).show();
            }
        });

The problem I am facing when I click Zoom In button, immediately it is not zooming in, after few clicks it completely zoom In to Maximum level

When I click the Zoom Out button, it is not zooming out progressively, after few clicks it completely zoom out 

Is there any alternate way for Zoom In / zoom out

I am using 2.4.7 version

implementation("com.github.pedroSG94:RootEncoder:2.4.7")

**Comment by pedroSG94** [comment]
Hello,

I think that the problem is the zoom values. 

First of all, I recommend you use getZoom method to get the actual zoom value instead of set a random value. 
After that, you can get maxZoom and minZoom and calculate the zoom step that you want to use. 1 is the min value for this. This could be a good implementation:
```java
    int iZoomLevel = rtmpCamera1.getZoom(); //get actual zoom
    //change this to increase sensibility. You can calculate sensibility using getMin and getMax zoom values
    int zoomStep = 1; 
    cameraZoomControls = findViewById(R.id.cameraZoomControls);
    cameraZoomControls.setActivated(true);
    cameraZoomControls.setIsZoomInEnabled(true);
    cameraZoomControls.setIsZoomOutEnabled(true);

    // Handle zoom-in click
    cameraZoomControls.setOnZoomInClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View view) {
            iZoomLevel = iZoomLevel + zoomStep;
            //max zoom is the limit
            if (iZoomLevel > rtmpCamera1.getMaxZoom()) iZoomLevel = rtmpCamera1.getMaxZoom();
            rtmpCamera1.setZoom(iZoomLevel);
            Toast.makeText(getApplicationContext(), "Zoom In Clicked!", Toast.LENGTH_SHORT).show();
        }
    });

   // Handle Zoom Out Click
    cameraZoomControls.setOnZoomOutClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View view) {
            iZoomLevel = iZoomLevel - zoomStep;
            //min zoom is the limit
            if (iZoomLevel < rtmpCamera1.getMinZoom()) iZoomLevel = rtmpCamera1.getMinZoom();
            rtmpCamera1.setZoom(iZoomLevel);
            Toast.makeText(getApplicationContext(), "Zoom Out Clicked!", Toast.LENGTH_SHORT).show();
        }
    });
```

**Reply by mylaimani** [reply]
Thanks it works.

**Comment by pedroSG94** [comment]
Other alternative is use zoom gestures:
```java
mCameraView.setOnTouchListener((v, event) -> {
            if (event.getPointerCount() > 1 && event.getAction() == MotionEvent.ACTION_MOVE) {
                rtmpCamera1.setZoom(event);
            }
            return true;
        });
```

**Reply by mylaimani** [reply]
I want to bring this to your attention, in my testing OpenGLView setOnTouchListner is not working in the version 2.5.0

It is working fine in version 2.4.7

**Reply by pedroSG94** [reply]
Hello,

If you are using camera2. This is a known issue already resolved.
You can use the grade provided here meanwhile:
https://github.com/pedroSG94/RootEncoder/issues/1575#issuecomment-2344355215
